db/migrate/20161003091412_add_exclusive_lock_indexes.rb

```diff
@@ -0,0 +1,7 @@
+class AddExclusiveLockIndexes < ActiveRecord::Migration
+  def change
+    add_index(:foreman_tasks_locks, [:name, :resource_type, :resource_id, :exclusive],
+              :name => 'index_foreman_tasks_locks_all_lock_properties')
```

Comment:
In theory, it shouldn't be necessary, but I would rather give it a try on some larger scale DB before replacing the one index fully.